### DISCUSSION TITLE
Configuration edit/import/export as yaml?

### DISCUSSION THREAD

**Discussion by maia** [jmcollin78/versatile_thermostat#155] [Q&A] [answered]
While it's great that versatile_thermostat has so many options, I'm sometimes overwhelmed with all the panels and options that are irrelevant for the mode I'm using ("over climate"), especially as I have about 10 climate entities, all with similar settings. Is there a way to edit/import/export the config of an entity in yaml mode instead of in the UI? That way I could e.g. quickly change the outdoor temperature sensor for all 10 entities by search&replace. Also people could then easily post their current config here, in case there are some issues they want to discuss.

**Comment by jmcollin78** [comment, marked as answer]
As is there is no way to do that simply. What is weird, is that it was the case before I use the graphical ConfigFlow. All configuration was in the configuration.yaml muche easier to configure on a "bulk way".

**Comment by maia** [comment]
Oh, so that's a general limitation of Home Assistant? So let's hope they'll add that feature in 2024.

**Comment by jmcollin78** [comment]
HA reads its configuration from .storage/core.config_entries file. This is the HA standard. At startup, HA creates devices and entities by reading this file.

It is possible to directly change the parameters in this file (i have done it myself). But:
1. always make a backup. If you write something wrong the system could not be able to start,
2. HA is writing this file each file you change something in one integration configuration,
3. you need to restart or at least reload the integration. There is a service for that or it is possible in Development Tools / yaml to reload Versatile Thermostat integration (but I never try this).

This is a way to do what you want. A little risky but possible.

**Reply by maia** [reply]
Yeah, thanks, I've seen that possibility. It's unfortunately nothing people will easily do when they want to share their configuration. Let's see if future changes in HA open new possibilities.
